src/cmd/support_archive/logs.go

```diff
@@ -0,0 +1,125 @@
+package support_archive
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const logCollectorName = "logCollector"
+
+type logCollector struct {
+	collectorCommon
+
+	context context.Context
+	pods    v1.PodInterface
+}
+
+func newLogCollector(context context.Context, log logr.Logger, supportArchive tarball, pods v1.PodInterface) collector { //nolint:revive // argument-limit doesn't apply to constructors
+	return logCollector{
+		collectorCommon: collectorCommon{
+			log:            log,
+			supportArchive: supportArchive,
+		},
+		context: context,
+		pods:    pods,
+	}
+}
+
+func (collector logCollector) Do() error {
+	logInfof(collector.log, "Starting log collection")
+
+	podList, err := collector.getPodList()
+	if err != nil {
+		return err
+	}
+
+	podGetOptions := metav1.GetOptions{}
+
+	for _, podItem := range podList.Items {
+		pod, err := collector.pods.Get(collector.context, podItem.Name, podGetOptions)
+		if err != nil {
+			logErrorf(collector.log, err, "Unable to get pod info for %s", podItem.Name)
+		} else {
+			collector.collectPodLogs(pod)
+		}
+	}
+	return nil
+}
+
+func (collector logCollector) Name() string {
+	return logCollectorName
+}
+
+func (collector logCollector) getPodList() (*corev1.PodList, error) {
+	listOptions := metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "pod",
+		},
+		LabelSelector: "app.kubernetes.io/name=dynatrace-operator",
+	}
+	podList, err := collector.pods.List(collector.context, listOptions)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return podList, nil
+}
+
+func (collector logCollector) collectPodLogs(pod *corev1.Pod) {
+	for _, container := range pod.Spec.Containers {
+		podLogOpts := corev1.PodLogOptions{
+			Container: container.Name,
+			Follow:    false,
+		}
+		collector.collectContainerLogs(pod, container, podLogOpts)
+
+		podLogOpts.Previous = true
+		collector.collectContainerLogs(pod, container, podLogOpts)
+	}
+}
+
+func (collector logCollector) collectContainerLogs(pod *corev1.Pod, container corev1.Container, logOptions corev1.PodLogOptions) {
+	req := collector.pods.GetLogs(pod.Name, &logOptions)
+	if req == nil {
+		logErrorf(collector.log, errors.Errorf("Invalid nil pointer returned instead of log stream for pod %s, container %s", pod.Name, container.Name), "")
+		return
+	}
+
+	podLogs, err := req.Stream(collector.context)
+
+	switch {
```

Comment:
very small nitpick: could be an `if {} else if {}` 😛

Reply:
I was waiting for this comment and very surprised it didn't come up earlier 😛